libmscore/line.cpp

```diff
@@ -989,7 +989,7 @@ void SLine::writeProperties(XmlWriter& xml) const
       //
       // write user modified layout
       //
-      qreal _spatium = spatium();
+      qreal _spatium = score()->spatium();
```

Comment:
This probably fixes the situation for `offset` but may break it for `off2`, it uses exactly local `spatium()` on score reading:
https://github.com/musescore/MuseScore/blob/3f911902996b68045f2b26a0eb1353910c16e4d9/libmscore/line.cpp#L61-L63
Maybe both these properties should be switched to using the same spatium value in case it doesn't spoil the existing files.

Reply:
Good catch, you are absolutely correct.

Since things are already quite messed up for existing files, I will switch this to using global spatium for off2 as well n read.  Will only affect scores that have small staves and diagonal lines, and they are likely to be messed up because of the offset bug that this commit was fixing.